### DISCUSSION TITLE
about BackgroundService.StartAsync

### DISCUSSION THREAD

**Discussion by hongsion** [dotnet/runtime#50673] [Q&A]
between line 39 - 45,

why if (_executingTask.IsCompleted), it return _executingTask instead of Task.CompletedTask?

to me looks correct behaviour should be as below:

if(_executingTask.IsCompleted)
{
     return Task.CompletedTask;
}

return _executingTask;

**Comment by davidfowl** [comment]
Because the executing task represents the lifetime of the task, not the lifetime of starting the task. The Start task represents the lifetime of the start operation, not the entire background service
